src/test/java/com/github/splendor_mobile_game/game/model/UserTests.java

```diff
@@ -0,0 +1,100 @@
+package com.github.splendor_mobile_game.game.model;
+
+import com.github.splendor_mobile_game.game.enums.CardTier;
+import com.github.splendor_mobile_game.game.enums.TokenType;
+import com.github.splendor_mobile_game.game.exceptions.NotEnoughTokensException;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class UserTests {
+    private final String userName = "USER";
+    private final String userUuid = "f8c3de3d-1fea-4d7c-a8b0-29f63c4c3454";
+    private final int userConnectionHashCode = 100000;
+
+    @Test
+    public void buyCardInsufficientTokensTest() {
+        User user = new User(UUID.fromString(this.userUuid), this.userName, this.userConnectionHashCode);
+        Card card = new Card(CardTier.LEVEL_1, 10,3,0,0,0,0, TokenType.EMERALD);
+
+        Throwable throwable = assertThrows(NotEnoughTokensException.class, () -> user.buyCard(card));
+        assertEquals("You don't have enough tokens to buy this card", throwable.getMessage());
```

Comment:
It's risky to assert the message. I would discourage that. On the other hand, asserting the type of the thrown exception is appropriate. If a developer decides to change the name of the exception, they would likely use refactoring tools to update all occurrences of it. However, if they change the message to something different, this change would not be reflected in other places, as string contents are not checked by the type checker of Java IntelliSense.

I propose removing checking message.